frappe/desk/search.py

```diff
@@ -175,18 +182,40 @@ def get_std_fields_list(meta, key):
 			if d.strip() not in sflist:
 				sflist.append(d.strip())
 
-	if meta.title_field and meta.title_field not in sflist:
-		sflist.append(meta.title_field)
-
 	if key not in sflist:
 		sflist.append(key)
 
 	return sflist
 
-def build_for_autosuggest(res):
+def get_title_field(meta, formatted_fields):
+	title_field = meta.title_field if meta.title_field else None
+	show_title_field_in_link = meta.show_title_field_in_link if meta.show_title_field_in_link else None
+
+	if title_field and show_title_field_in_link:
+		field = "`tab{0}`.{1} as `label`".format(meta.name, title_field)
+	else:
+		field = "NULL as `label`"
+
+	formatted_fields.insert(1, field)
```

Comment:
Do not mutate external variable if the function name is starting with "`get_`", it makes function name misleading.

In this case, you can simply return the label query